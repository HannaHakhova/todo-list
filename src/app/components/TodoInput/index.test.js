/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import {shallow} from 'enzyme';
import TodoInput from './index';

const component = shallow(<TodoInput handleSubmit={() => {}} />);

describe('TodoInput', () => {
    it('matches expected snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('calls handleSubmit when form is submitted and sets empty state after call', () => {
        const handleSubmit = jest.fn();
        const wrapper = shallow(<TodoInput handleSubmit={handleSubmit} />);
        wrapper.setState({title: 'test'});

        wrapper.find('Form').simulate('submit');

        expect(handleSubmit).toBeCalledWith('test');
        expect(wrapper.state().title).toEqual('');
    });

    it('handles change on some input', () => {
        component.find('FormInput').simulate('change', {target: {value: 'Test2'} });

        expect(component.state().title).toEqual('Test2');
    });
});