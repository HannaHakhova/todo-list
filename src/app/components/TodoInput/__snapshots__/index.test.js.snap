// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput matches expected snapshot 1`] = `
<Form
  as="form"
  onSubmit={[Function]}
>
  <FormGroup
    widths="equal"
  >
    <FormInput
      action="Add"
      as={[Function]}
      control={[Function]}
      onChange={[Function]}
      placeholder="Add todo..."
      size="large"
      type="text"
      value=""
    />
  </FormGroup>
</Form>
`;
